File: src/dataset/cpp/knapsack_exercism-toemm/claude/ClaudeSonnet4_knapsack_v3.cpp
#include "knapsack.h"

#include <algorithm>
#include <vector>

namespace knapsack {

int maximum_value(int max_weight, std::vector<Item> items) {
    if (items.empty() || max_weight <= 0) {
        return 0;
    }

    const int n = items.size();
    std::vector<int> prev(max_weight + 1, 0);
    std::vector<int> curr(max_weight + 1, 0);

    for (int i = 0; i < n; ++i) {
        const int weight = items[i].weight;
        const int value = items[i].value;

        for (int w = 0; w <= max_weight; ++w) {
            if (weight > w) {
                curr[w] = prev[w];
            } else {
                curr[w] = std::max(prev[w], prev[w - weight] + value);
            }
        }
        prev.swap(curr);
    }

    return prev[max_weight];
}

}  // namespace knapsack

File: src/dataset/python/knapsack_exercism-fruit-in/claude/ClaudeSonnet4_knapsack_v3.py
def maximum_value(maximum_weight, items):
    dp = [0] * (maximum_weight + 1)

    for item in items:
        weight = item["weight"]
        value = item["value"]
        
        for w in range(maximum_weight, weight - 1, -1):
            dp[w] = max(dp[w], dp[w - weight] + value)

    return dp[maximum_weight]